### PR TITLE
add comments

### PR DESCRIPTION
**Pull request by fluidvanadium**
Contains some comments about BlockManagementData discovered this branch in one of my worktrees.
Contains descriptions of transactions.rs and its sub-modules.

### CODE REVIEW

File: zingolib/src/blaze/block_management_reorg_detection.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ type Node<D> = <<D as DomainWalletExt>::WalletNote as ShieldedNoteInterface>::No
 const ORCHARD_START: &str = "000000";
 /// The data relating to the blocks in the current batch
 pub struct BlockManagementData {
-    // List of all downloaded blocks in the current batch and
+    // List of all downloaded Compact Blocks in the current batch and
```
Comment (Oscar-Pepper) [inline zingolib/src/blaze/block_management_reorg_detection.rs:39]: not sure what the capitals are for?

File: zingolib/src/blaze/block_management_reorg_detection.rs [hunk 2]
```diff
@@ -446,17 +446,22 @@ impl BlockManagementData {
         self.wait_for_block(after_height).await;
 
         {
-            // Read Lock
+            // Read Lock for
```
Comment (Oscar-Pepper) [inline zingolib/src/blaze/block_management_reorg_detection.rs]: comment seems incomplete

File: zingolib/src/blaze/block_management_reorg_detection.rs [hunk 3]
```diff
@@ -446,17 +446,22 @@ impl BlockManagementData {
...
             let blocks = self.blocks_in_current_batch.read().await;
+            // take the height of the first/highest block in the batch and subtract target height to get index into the block list
             let pos = blocks.first().unwrap().height - after_height;
             match nf {
                 PoolNullifier::Sapling(nf) => {
                     let nf = nf.to_vec();
 
+                    // starting with a block and searching until the top of the batch
```
Comment (Oscar-Pepper) [inline zingolib/src/blaze/block_management_reorg_detection.rs:458]: I think the code is much more readable when not broken up by many comments. mabye describe the block in one comment above?

File: zingolib/src/blaze/update_notes.rs [hunk 4]
```diff
@@ -113,6 +113,7 @@ impl UpdateNotes {
                         .await
                     {
                         //info!("Note was spent, just add it as spent for TxId {}", txid);
+                        // we got the height the nullifier was spent at. now, we go back to the index because todO actually we needed the CompactTx, not just the height
```
Comment (Oscar-Pepper) [inline zingolib/src/blaze/update_notes.rs]: comment seems incomplete

**Review by Oscar-Pepper — CHANGES_REQUESTED** [review]

File: zingolib/src/blaze/block_management_reorg_detection.rs (hunk 1 above)
Comment (fluidvanadium) [reply zingolib/src/blaze/block_management_reorg_detection.rs:39]: because its a struct name.

File: zingolib/src/blaze/update_notes.rs (hunk 4 above)
Comment (fluidvanadium) [reply zingolib/src/blaze/update_notes.rs]: oops, these were unfinished! im so glad i got a chance to finish them

File: zingolib/src/blaze/block_management_reorg_detection.rs (hunk 2 above)
Comment (fluidvanadium) [reply zingolib/src/blaze/block_management_reorg_detection.rs]: completed

**Review by zancas — APPROVED** [review]

**Pull request merged**